allennlp/data/dataset_readers/penn_tree_bank.py

```diff
@@ -0,0 +1,173 @@
+
+from collections import OrderedDict
+from typing import Dict, List
+import logging
+import os
+
+from overrides import overrides
+# NLTK is so performance orientated (ha ha) that they have lazy imports. Why? Who knows.
+from nltk.corpus.reader.bracket_parse import BracketParseCorpusReader # pylint: disable=no-name-in-module
+from nltk.tree import Tree
+
+from allennlp.common import Params
+from allennlp.common.file_utils import cached_path
+from allennlp.common.tqdm import Tqdm
+from allennlp.data.dataset_readers.dataset_reader import DatasetReader
+from allennlp.data.fields import TextField, SpanField, SequenceLabelField, ListField, Field
+from allennlp.data.instance import Instance
+from allennlp.data.token_indexers import TokenIndexer, SingleIdTokenIndexer
+from allennlp.data.tokenizers import Token
+from allennlp.data.dataset_readers.dataset_utils.span_utils import enumerate_spans
+
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+@DatasetReader.register("ptb_trees")
+class PennTreeBankDatasetReader(DatasetReader):
+    """
+    Reads constituency parses from the WSJ part of the Penn Tree Bank from the LDC.
+
+    Parameters
+    ----------
+    token_indexers : ``Dict[str, TokenIndexer]``, optional (default=``{"tokens": SingleIdTokenIndexer()}``)
+        We use this to define the input representation for the text.  See :class:`TokenIndexer`.
+        Note that the `output` tags will always correspond to single token IDs based on how they
+        are pre-tokenised in the data file.
+    """
+    def __init__(self,
+                 token_indexers: Dict[str, TokenIndexer] = None) -> None:
+        super().__init__(lazy=False)
```

Comment:
I think you should take a `lazy` parameter and pass it up, instead of only allowing `False` here.